### ISSUE TITLE
Invalid array length Allocation failed, OOM/Coredump

### ISSUE THREAD

**Issue by johnalotoski** [input-output-hk/cardano-graphql#268]
### Description
* Cardano graphql is randomly core dumping due to invalid array length allocation
* Two example events are attached

### Environment
cardano-graphql repo: `refs/tags/2.0.0`

#### Platform

- [*] Linux / Other: NixOS 20.03
- [*] Node.js

### Steps to reproduce the bug
This error is occurring randomly with a frequency of once every couple of days

### Logs/Backtraces
[cardano-graphql-core-dump-2020-07-26.zip](https://github.com/input-output-hk/cardano-graphql/files/5007291/cardano-graphql-core-dump-2020-07-26.zip)
[cardano-graphql-core-dump-2020-07-31.zip](https://github.com/input-output-hk/cardano-graphql/files/5007292/cardano-graphql-core-dump-2020-07-31.zip)

**Comment by SebastienGllmt** [comment]
This same bug just took down Yoroi servers for a while
```
Aug 02 11:18:58 backend2 cardano-graphql-start[7666]: FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory
```

**Comment by rhyslbw** [comment]
@SebastienGllmt  what environment are you running the service in ?

**Comment by rhyslbw** [comment]
I've upgraded the `Db` client to use the PostgreSQL notifications being sent from `cardano-db-sync` to trigger schema migrations, instead of polling the db, which is where I suspect the leak is occurring. A database rebuild is required, which my CI is doing now.

**Comment by rhyslbw** [comment]
Expecting this to be resolved by #275
